rfc-164-retire-email-alert-monitoring-as-a-separate-app.md

```diff
@@ -0,0 +1,76 @@
+# Retire email-alert-monitoring as a separate app
+
+## Summary
+
+Replace the current email-alert-monitoring system with a task built into email-alert-api using the Notify API rather than the current Gmail/App based system, giving us a more stable and testable system and allowing us to retire a repo.
+
+## Problem
+
+email-alert-monitoring is a stand-alone app run as a cronjob (currently on Jenkins, although it's being moved over to EKS). It's designed to confirm that medical and travel alert emails have been sent out on time, and supports an SLA. It's clearly critical that we have some way of knowing that these alerts have gone out, but the current system has a number of problems:
+
+- It relies on a slightly Heath Robinson collection of google accounts
+- Its method of matching alerts to emails is brittle
+- False alarms well outnumber valid alarms (which causes annoying on-call pages)
+- It's not entirely clear how actionable the valid alarms are
+
+## Proposal
+
+We should retire email-alert-monitoring as a separate application, and instead build a system into email-alert-api that can be called by a cron job (as is currently the case for the separate app).
+
+Currently it looks like we're not tracking the id for emails sent out via notify, but we do store email information for one day. By storing the notify ID, either for all emails (easier) or just for alert emails, we could then [query notify as to their current status]. Our cron job could query the current status for the relevant alert emails, and if _one_ email is in the "delivered" state, we have roughly as much proof that they're being delivered as we currently do.
+
+As a side-effect of adding this, we could also potentially query every alert email in the given period to determine how many of our alerts on average are getting delivered (and perhaps identify recipients who no longer exist).
+
+This isn't quite as end-to-end as the current system (if something was delivered, does that mean it's actually in a given person's inbox?), but since any record that Notify deems delivered is at the absolute boundaries of anything actionable by GOV.UK, there seems to be little to gain by tracking the alerts further than that.
```

Comment:
I agree with this thinking. But I think it's also worth keeping the existing `govuk_email_check@digital.cabinet-office.gov.uk` inbox and subscriptions. That way, if we get reports from departments/users that emails aren't being received, 2nd line can check if Notify is reporting that emails are being sent, and can _also_ verify that they've physically seen an email land in the inbox. We should just remove the monitoring around the account, as you're proposing 👍

Reply:
I definitely think we should keep the existing subscriptions, but - we can move them from the current email address to an address backed by a google group rather than a full gmail account. We'd still be able to manually confirm that emails have got into the group, but:
- google groups are free, gmail accounts cost money, and
- google groups can be easily joined by any dev, and don't require an additional password hunt/login (reducing use of govuk-secrets)

I'll add this in.

Reply:
How does this help us? I had a discussion with Keith around some of this that led to us finding out more about how Notify satisfies itself that things have been delivered. If gmail/google groups is telling Notify "yep, got that" and we don't see the message, what lines of investigation does that actually open for us? If rando email server tells Notify that it got a message, rando user doesn't see it, and yet the gmail journey is fine and we've got one, again, what lines of enquiry are actually open? Is there definitely a value add in keeping a delivery chain open for this? Unless the users experiencing problems are also having their subs sent to a google group I'm not sure this adds much to the assurance.